velox/exec/tests/AssignUniqueIdTest.cpp

```diff
@@ -166,3 +179,66 @@ TEST_F(AssignUniqueIdTest, taskUniqueIdLimit) {
       AssertQueryBuilder(plan).copyResults(pool()),
       "(16777216 vs. 16777216) Unique 24-bit ID specified for AssignUniqueId exceeds the limit");
 }
+
+TEST_F(AssignUniqueIdTest, barrier) {
+  std::shared_ptr<const RowType> rowType{
+      ROW({"c0", "c1"}, {BIGINT(), INTEGER()})};
+
+  // create tmp table
+  const int numSplits{5};
+
+  auto tempFiles = [&]() {
+    std::vector<RowVectorPtr> vectors;
+    std::vector<std::shared_ptr<TempFilePath>> tempFiles;
+
+    const int numRowsPerSplit{100};
+    for (int32_t i = 0; i < numSplits; ++i) {
+      auto vector = std::dynamic_pointer_cast<RowVector>(
+          BatchMaker::createBatch(rowType, numRowsPerSplit, *pool_));
```

Comment:
BatchMaker is deprecated. Would you use VectorMaker or better yet makeXxx methods from the base class?